src/relay/backend/contrib/cmsisnn/relay_to_tir.cc

```diff
@@ -587,10 +587,11 @@ class RelayToTIRVisitor : public MixedModeMutator {
   }
 
   void EmitMul(const GlobalVar& global_var, const Expr& expr) {
-    int32_t output_min = std::numeric_limits<int8_t>::min();
-    int32_t output_max = std::numeric_limits<int8_t>::max();
     const auto& pattern = ParseBinaryElementwiseOpClipPattern(expr);
     Call mul_call = pattern.binary_op;
+    const auto bit_width = mul_call->type_as<TensorTypeNode>()->dtype.bits();
+    int32_t output_min = -(1 << (bit_width - 1));
+    int32_t output_max = (1 << (bit_width - 1)) - 1;
```

Comment:
nit: Can we save`std::numeric_limits()` as part of the class and use it where needed? Looks more standard than bit based calculations.

Reply:
Yeah good shout. I've tried to replace some code duplication throughout this file, so let me konw if that's what you had in mind :D